addons/sale_project/models/sale_order.py

```diff
@@ -163,7 +163,7 @@ def action_view_task(self):
             action['res_id'] = self.tasks_ids.id
         # set default project
         default_line = next((sol for sol in self.order_line if sol.product_id.detailed_type == 'service'), self.env['sale.order.line'])
-        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id
+        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id or self.tasks_ids.project_id[-1].id
```

Comment:
can't we have 0 task (from above it is written "else: # 1 or 0 tasks -> form view"). then you would get an "IndexError: list index out of range" error with [-1] of an empty recordset, it should be eg `[:1]` if you want the first element (or empty recordset) or `[-1:]` if you wnat the last element (or empty recordset):

```suggestion
        default_project_id = default_line.project_id.id or self.project_id.id or self.project_ids[:1].id or self.tasks_ids.project_id[:1].id
```